Split sqllite and sql server migrations

HaoK opened the pull request:
Part of https://github.com/aspnet/templating/issues/309

@seancpeters does this look right?  We already have a UseLocalDb symbol we can use...  I tried this and I don't think the rename/exclusion is working as I have it, since both sets of files are showing up still...Can you check if there something obviously wrong?

seancpeters commented:
@HaoK - I believe that the problem is because template symbols are case-sensitive. In the symbols section, there is UseLocalDB, but in the source modifier conditions it's UseLocalDb.

HaoK commented:
@seancpeters that was it!  After talking with @ajcvickers I moved things around a bit, so there's now a Data\SqlLite and Data\SqlServer which contain the 3 migration files, and they get renamed to Data\Migrations.  

This should be ready for review now, the functional mvc/razor tests are passing

ajcvickers commented on src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/Data/SqlLite/ApplicationDbContextModelSnapshot.cs:
```diff
@@ -12,20 +12,66 @@
+
+                    b.HasIndex("NormalizedName")
+                        .IsUnique()
+#if (UseLocalDB)
```
Shouldn't need this anymore, right?

HaoK replied:
Oops yeah good catch I'll remove these two

ajcvickers commented on src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/Data/SqlServer/ApplicationDbContextModelSnapshot.cs:
```diff
@@ -12,20 +12,66 @@
+
+                    b.HasIndex("NormalizedName")
+                        .IsUnique()
+#if (UseLocalDB)
```
Likewise.

Eilon commented on src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/Data/SqlLite/ApplicationDbContextModelSnapshot.cs:
```diff
@@ -12,20 +12,66 @@
             modelBuilder
 #if (UseLocalDB)
-                .HasAnnotation("ProductVersion", "2.1.0-preview1")
+                .HasAnnotation("ProductVersion", "2.1.0-preview1-28193")
```
@ajcvickers / @bricelam - just checking, the version here is OK? I know we've discussed this a billion times but I can never remember, sorry!

ajcvickers replied:
@Eilon It doesn't need to match anything so this is fine for preview, but we should update it before RTM ships--I filed https://github.com/aspnet/templating/issues/353 for this and assigned it to @HaoK

Eilon commented on src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/Data/SqlLite/ApplicationDbContextModelSnapshot.cs:
```diff
@@ -241,7 +240,8 @@ protected override void BuildTargetModel(ModelBuilder modelBuilder)
+
 #pragma warning restore 612, 618
-        }
+         }
```
How did this extra space get here? The indentation is now off. Is this a bug in the migration generation logic itself?

HaoK replied:
These current files are still the result of the previous hand merge/edits.  The next PR will add the batch files that can be used to generate these for check in purposes, and I'll regenerate the migrations with the new batch files.

Eilon replied:
Ah OK so I'll ignore for now. Thanx!

Eilon commented on src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/Data/SqlServer/ApplicationDbContextModelSnapshot.cs:
```diff
@@ -241,7 +240,8 @@ protected override void BuildTargetModel(ModelBuilder modelBuilder)
+
 #pragma warning restore 612, 618
-        }
+         }
```
See earlier comment.

seancpeters commented on src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/.template.config/template.json:
```diff
@@ -78,6 +78,24 @@
             "Extensions/AzureAdB2C*.cs"
           ]
+        },
+        {
```
There appears to be a conflict between these new modifiers and this existing modifier:
```
       {
          "condition": "(!IndividualLocalAuth)",
          "exclude": [
            "Data/**",
            "Areas/**"
          ]
        },
```
If a user chooses auth = IndividualB2C, then that makes IndividualLocalAuth = false, causing the above condition to be true, which excludes the Data/ directory. 

These new modifiers, conditioned on UseLocalDB, want to include some of the content under the Data/ directory, but it's been excluded.

The existing modifier which excludes the Data/ dir might need to be revisited with this in mind.

seancpeters replied:
Looking at the rest of this review, I noticed that for StarterWeb CSharp, the modifier to exclude Data/ & Areas/ is conditioned on "(!IndividualAuth)", which might be the desired change here.

HaoK replied:
Well it shouldn't matter as the rename won't do anything if the top level Data was all excluded right?  But seems harmless to add an additional IndividualLocalAuth check for these new conditions...

seancpeters replied:
Correct, the rename won't do anything if the Data folder is already excluded. 

But it seems to be the case that with auth = IndividualB2C, we'd want to include the appropriate migration files - based on the UseLocalDB value. If this is incorrect, then it's probably fine as is.

HaoK replied:
IndividualB2C = true means IndividualLocalAuth = false...  so the migrations should and are excluded.  I'm happy to add the IndividualLocalAuth check if you think that makes it clearer in the long run...  its not exactly obvious how all this stuff plays together at all, so even redundant checks have some value for clarity in the future

seancpeters replied:
I don't feel too strongly either way about adding the additional check.

The reason brought this up originally is because the description on the UseLocalDB parameter indicates that it is applicable for both Individual & IndividualB2C auth types. With that in mind, I assumed that migrations would be desired for B2C.

HaoK replied:
These are IndividualAuth specific migrations so that's why

HaoK commented:
Updated cleaning up the ApplicationDbContextSnapshot if defs, and added extra IndividualLocalAuth checks to the conditions, look good @seancpeters ?  The files on disk are the same either way, so functionally I don't think it makes a difference
